boot/initramfs.go

```diff
@@ -0,0 +1,69 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package boot
+
+import (
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/snap"
+)
+
+// TODO:UC20: add install/recover mode mounts code here too?
+
+// InitramfsRunModeSelectSnapsToMount returns a map of the snap paths to mount
+// for the specified snap types.
+func InitramfsRunModeSelectSnapsToMount(
+	typs []snap.Type,
+	modeenv *Modeenv,
+) (map[snap.Type]snap.PlaceInfo, error) {
+	var sn snap.PlaceInfo
+	var err error
+	m := make(map[snap.Type]snap.PlaceInfo)
+	for _, typ := range typs {
+		// TODO: it would be nice to use an interface here since the methods are
+		// already organized this way...
```

Comment:
the interface doesn't help with the construction, no? so it would just be the method?  that can be covered with a function var no?

```
var  select fun()  (snap.PlaceInfo, error)

select := bs....

select := bs....
```

am I missing something more?

Reply:
well it would be nice if the interface could also help with the construction. at this point I don't even remember how I initially solved this because I did originally have this as an interface, but doing as you show with a var for a func is fine and reduces some of the duplicative code here